lib/better_html/tokenizer/base_erb.rb

```diff
@@ -0,0 +1,78 @@
+require 'erubi'
+require_relative 'token'
+require_relative 'location'
+
+module BetterHtml
+  module Tokenizer
+    class BaseErb < ::Erubi::Engine
+      REGEXP_WITHOUT_TRIM = /<%(={1,2}|%)?(.*?)()?%>([ \t]*\r?\n)?/m
+      STMT_TRIM_MATCHER = /\A(-|#)?(.*?)(-)?\z/m
+      EXPR_TRIM_MATCHER = /\A(.*?)(-)?\z/m
+
+      attr_reader :tokens
+      attr_reader :current_position
+
+      def initialize(document)
+        @document = document
+        @tokens = []
+        @current_position = 0
+        super(document, regexp: REGEXP_WITHOUT_TRIM, trim: false)
+      end
+
+      private
+
+      def append(text)
+        @current_position += text.length
+      end
+
+      def add_code(code)
+        _, ltrim_or_debug, code, rtrim = *STMT_TRIM_MATCHER.match(code)
+        ltrim = ltrim_or_debug if ltrim_or_debug == '-'
+        indicator = ltrim_or_debug if ltrim_or_debug == '#'
```

Comment:
Also rename `ltrim_or_debug` to `ltrim_or_comment`?

Reply:
👍